src/index.js

```diff
@@ -1,32 +1,136 @@
+    this.index = index;
+    this.completed = completed;
+    this.description = description;
+  }
+}
+
+const addTask = (description) => {
+  const storedTasks = JSON.parse(localStorage.getItem('tasks'));
+  if (storedTasks === null) {
+    allTasks = [];
+  } else {
+    allTasks = JSON.parse(localStorage.getItem('tasks'));
+  }
+  const newTask = new Task(allTasks.length, false, description);
+  allTasks.push(newTask);
+  localStorage.setItem('tasks', JSON.stringify(allTasks));
+  window.location.reload();
+};
+
+const editTask = (index, description) => {
+  const task = allTasks.find((task) => task.index === index);
+  task.description = description;
+  localStorage.setItem('tasks', JSON.stringify(allTasks));
+};
+
+const toggleTask = (index, completed) => {
+  const task = allTasks.find((task) => task.index === index);
+  task.completed = completed;
+  localStorage.setItem('tasks', JSON.stringify(allTasks));
+};
 
-const tasks = rawTasks.sort((a, b) => a.index - b.index);
+const removeTask = () => {
+  const completedTasks = [];
+  allTasks.forEach((task) => {
+    if (task.completed === false) {
+      completedTasks.push(task);
+    }
+  });
+  for (let i = 0; i < completedTasks.length; i += 1) {
+    completedTasks[i].index = i;
+  }
+  allTasks = completedTasks;
+  localStorage.setItem('tasks', JSON.stringify(allTasks));
+  window.location.reload();
+};
+
+const paper = document.getElementById('paper');
+
+if (allTasks.length === 0) {
+  tasks = [];
+  if (paper.childNodes.length < 4) {
+    const footer = document.getElementById('remove');
+    paper.removeChild(footer);
+  }
+} else {
+  tasks = allTasks.sort((a, b) => a.index - b.index);
+  const footer = document.createElement('footer');
+  footer.id = 'remove';
+  footer.innerHTML = 'Clear all completed';
+  footer.onclick = removeTask;
+  paper.appendChild(footer);
+}
+
+const add = document.getElementById('add');
+add.addEventListener('keydown', (e) => {
+  if (e.key === 'Enter') {
+    addTask(add.value);
+    add.value = '';
+  }
+});
 
 tasks.forEach((task) => {
   const li = document.createElement('li');
   const div = document.createElement('div');
   div.style.cssText = 'display: flex; padding-left: 10px;';
-  const input = document.createElement('input');
-  input.type = 'checkbox';
-  input.style.cssText = 'cursor: pointer;';
+  const checkbox = document.createElement('input');
+  checkbox.type = 'checkbox';
+  checkbox.style.cssText = 'cursor: pointer;';
+  checkbox.checked = task.completed;
+  checkbox.addEventListener('change', () => {
+    if (checkbox.checked) {
+      toggleTask(task.index, true);
+      li.style.cssText = 'text-decoration: line-through; background-color: gainsboro;';
+    } else {
+      toggleTask(task.index, false);
+      li.style.cssText = 'text-decoration: none; background-color: white;';
+    }
+  });
   const p = document.createElement('p');
   p.innerHTML = task.description;
+  // console.log(task.description)
```

Comment:
- [OPTIONAL ] Do not commit `console.log` to your repo. It’s ugly, it kills performance and it can make confidential data visible to anyone using the browser tools to look at your website. kindly remove this 😄